### PR TITLE
token as auth for scheduler

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [18075](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18075)

### Description

Use a uuid token for identification on the jobs callback and removed authentication for the route, requiring a token to execute the job.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging-api/routes/messages/index.ts [hunk 1]
```diff
@@ -330,16 +338,21 @@ export default async function messages(app: FastifyInstance) {
     },
   );
 
-  app.get<{ Querystring: { search?: string } }>(
+  app.get<{
+    Querystring: { search?: string; offset?: number; limit?: number };
```
Comment (SamSalvatico) [inline apps/messaging-api/routes/messages/index.ts:342]: Please use `{ search?: string;} & PaginationParams `

File: apps/messaging-api/services/messages/messaging.ts [hunk 2]
```diff
@@ -232,32 +233,37 @@ export function newMessagingService(pool: Pool): Readonly<MessagingService> {
...
       }
 
       const jobs: {
         jobId: string;
         userId: string;
         entityId: string;
+        token: string;
       }[] = [];
       try {
+        console.log(values, valueArgs);
```
Comment (SamSalvatico) [inline apps/messaging-api/services/messages/messaging.ts:256]:
```suggestion
 
```

File: apps/messaging-api/types/schemaDefinitions.ts [hunk 3]
```diff
@@ -58,8 +58,8 @@ export const CreateMessageSchema = Type.Composite([
 export type CreateMessage = Static<typeof CreateMessageSchema>;
 
 export const PaginationParamsSchema = Type.Object({
-  offset: Type.Optional(Type.Integer({ default: 0 })),
-  limit: Type.Optional(Type.Integer({ default: 20 })),
+  offset: Type.Optional(Type.Integer({ default: 0, minimum: 0 })),
+  limit: Type.Optional(Type.Integer({ default: 20, minimum: 1 })),
```
Comment (SamSalvatico) [inline apps/messaging-api/types/schemaDefinitions.ts:62]: 🙇

File: apps/messaging-api/routes/messages/index.ts [hunk 4]
```diff
@@ -204,6 +205,13 @@ export default async function messages(app: FastifyInstance) {
         throw new ServerError(errorKey, "no user id on request");
       }
 
+      if (!req.userData?.organizationId) {
+        throw new ServerError(
```
[the remark this answers is not in the document]
Comment (Thurfjell) [reply apps/messaging-api/routes/messages/index.ts:206]: Yeah that's a good one.

**Pull request merged**